File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import itertools

def isPrime(n):
    if n<2: return False
    if n==2: return True
    if n%2==0: return False
    for i in range(3, n//2+1, 2):
        if n%i==0: return False
    return True

def solution(numbers):
    numberList=list(numbers)
    nums=set()
    
    for i in range(1, len(numberList) + 1):
        for perm in itertools.permutations(numberList, i):
            num_str = ''.join(perm).lstrip('0')
            if num_str:  # 빈 문자열을 방지
                nums.add(int(num_str))
    
    primes = [n for n in nums if isPrime(n)]
            
    return len(primes)

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function getPermutation(permu, rests, output) {
    if (permu.length > 0) output.push(Number(permu.join('')));
    if (rests.length === 0) return;
    
    rests.forEach((v, i) => {
        const rest = [...rests.slice(0, i), ...rests.slice(i + 1)]
        getPermutation([...permu, v], rest, output);
    })
}

function isPrime(n) {
    if (n < 2) return false;
    if (n === 2) return true;
    if (n % 2 ===0) return false;
    for (let i = 3; i < Math.sqrt(n) + 1; i++) {
        if (n % i === 0) return false;
    }
    return true;
}

function solution(numbers) {
    let count = 0;
    let perm = []
    getPermutation([], numbers.split(''), perm);
    perm = [...new Set(perm)];
    
    for (let p of perm) {
        if (isPrime(p)) count++;
    }
    return count;
}